types/gatherer.d.ts

```diff
@@ -47,6 +47,8 @@ declare module Gatherer {
     url: () => Promise<string>;
     targetManager: {
       rootSession(): FRProtocolSession;
+      executionContexts(): Array<Crdp.Runtime.ExecutionContextDescription>;
+      mainFrameExecutionContexts(): Array<Crdp.Runtime.ExecutionContextDescription>;
```

Comment:
Although we only want to process main frames in global listeners, I don't want that to happen via a method called just `executionContexts` because that locks in the contract that this is only main frames (which could mess with potential plugins). Having the gatherer provide the main frame id is not possible for snapshots, so this logic is kept in target manager. Hence, an explicit `mainFrameExecutionContexts`... and adding `executionContexts` just-cuz.